libs/go/sia/gcp/meta/meta.go

```diff
@@ -0,0 +1,129 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package meta
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// GetData makes a http call to the local metadata end point and returns the metadata document as bytes
+func GetData(base, path string) ([]byte, error) {
+	headers := make(map[string]string)
+	headers["Metadata-Flavor"] = "Google"
+	return processHttpRequest(base, path, "GET", headers)
+}
+
+func processHttpRequest(base, path, method string, headers map[string]string) ([]byte, error) {
+	c := &http.Client{}
+	c.Timeout = 5 * time.Second
+	req, err := http.NewRequest(method, base+path, nil)
+	if err != nil {
+		return nil, err
+	}
+	if headers != nil {
+		for key, value := range headers {
+			req.Header.Set(key, value)
+		}
+	}
+	res, err := c.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	body, err := io.ReadAll(res.Body)
+	_ = res.Body.Close()
+	if err != nil {
+		return nil, err
+	}
+	if res.StatusCode == 200 {
+		return body, nil
+	}
+	return nil, fmt.Errorf("cannot get metadata, url: %q, headers: %v status code is %d", base+path, headers, res.StatusCode)
+}
+
+// GetRegion get current region from identity document
+func GetRegion(metaEndPoint string) string {
+	var region string
+	region = getRegionFromMeta(metaEndPoint)
+	if region == "" {
+		log.Println("No region information available. Defaulting to us-west1")
+		region = "us-west1"
+	}
+	return region
+}
+
+func getRegionFromMeta(metaEndPoint string) string {
+	var region string
+	log.Println("Trying to determine region from metadata server ...")
+	fullOutput, err := GetData(metaEndPoint, "/computeMetadata/v1/instance/zone")
+	if err == nil {
+		if idx := strings.LastIndex(string(fullOutput), "/"); idx > 0 {
+			zone := string(fullOutput[idx+1:])
+			if idx := strings.LastIndex(zone, "-"); idx > 0 {
+				region = zone[:idx]
+			}
+		}
+	}
+	return region
+}
+
+func GetDomain(metaEndpoint string) (string, error) {
+	domainBytes, err := GetData(metaEndpoint, "/computeMetadata/v1/project/attributes/domain")
+	if err != nil {
+		return "", err
+	}
+	return string(domainBytes), nil
+}
+
+func GetProject(metaEndpoint string) (string, error) {
+	projectBytes, err := GetData(metaEndpoint, "/computeMetadata/v1/project/project-id")
+	if err != nil {
+		return "", err
+	}
+	return string(projectBytes), nil
+}
+
+func GetService(metaEndpoint string) (string, error) {
+	serviceBytes, err := GetData(metaEndpoint, "/computeMetadata/v1/instance/service-accounts/default/email")
+	if err != nil {
+		return "", err
+	}
+	if idx := strings.Index(string(serviceBytes), "@"); idx > 0 {
+		service := string(serviceBytes[:idx])
+		return service, nil
+	}
+	return "", fmt.Errorf("unable to derive service name from metadata")
+}
+
+func GetProfile(metaEndpoint string) (string, error) {
+	profileBytes, err := GetData(metaEndpoint, "/computeMetadata/v1/instance/attributes/calypsoProfile")
```

Comment:
we had decided not to use calypso in any of the open source calls. it is referred as accessProfile in most other places so let's use that value here as well.